compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -669,6 +675,11 @@ LogicalResult workgroupToAIE(IRRewriter &rewriter,
   int dmaId = 0;
   WalkResult res = workgroupOp.walk<WalkOrder::PreOrder>([&](Operation *op) {
     return TypeSwitch<Operation *, WalkResult>(op)
+        .Case<AMDAIE::BdIdOp>([&](auto bdIdOp) {
+          // BD ID ops are purely used for retrieving information in other ops
+          // so don't convert to AIE dialect.
+          return WalkResult::advance();
```

Comment:
i didn't notice - how do they disappear? dce?

Reply:
In this pass, the whole IR is walked and used to construct the new `aie.device`. The old part of the IR is just deleted as a whole.